File: LeetCode/question1/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
    '''
    枚举
    '''
        for index, item in enumerate(nums):
            for i, t in enumerate(nums[index+1:]):
                if t + item == target:
                    return [index, i+index+1]
        return []
        
    def twoSum_plus(self, nums: List[int], target: int) -> list[int]:
    '''
    使用字典构成哈希表
    '''
        hashmap = {}
        for index, num in enumerate(nums):
            another_num = target - num
            if another_num in hashmap:
                return [hashmap[another_num], index]
            hashmap[num] = index
        return None

File: LeetCode/question1/solution.c
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    int i, j, flag = 0;
    int *return_val = NULL;
    for (i = 0; i < numsSize-1; i++) {
        int has_to_find = target - nums[i];
        for (j = i + 1; j < numsSize; j++) {
            if (nums[j] == has_to_find) {
                return_val = (int *) malloc(2*sizeof(int));
                return_val[0] = i;
                return_val[1] = j;
                *returnSize = 2;
                return return_val;
            }
        }
    }
    *returnSize = 0;
    return return_val;
}
